File: 11-ContainerWithMostWater/11-ContainerWithMostWater.c
// Last updated: 1/8/2026, 5:29:31 p.m.
int maxArea(int* height, int heightSize) {
    int i = 0, j = heightSize - 1, amount = 0, aux;

    while(i < j){
        if(height[j] > height[i]) aux = (j-i) * height[i++];
        else aux = (j-i) * height[j--];

        if(aux > amount)
            amount = aux;
    }

    return amount;
}

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 1/8/2026, 5:29:28 p.m.
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height)-1
        prev_area = 0

        while i < j:
            area = (j-i)*min(height[i], height[j])
            if area > prev_area: 
                prev_area = area

            if height[j] > height[i]: i += 1
            else: j -= 1
    
        return prev_area